include/net/net_context.h

```diff
@@ -769,6 +777,38 @@ int net_context_recv(struct net_context *context,
 int net_context_update_recv_wnd(struct net_context *context,
 				s32_t delta);
 
+enum net_context_option {
+	NET_OPT_PRIORITY = 1,
+};
+
+/**
+ * @brief Set an connection option for this context.
+ *
+ * @param context The network context to use.
+ * @param option Option to set
+ * @param value Option value
+ * @param len Option length
+ *
+ * @return 0 if ok, <0 if error
+ */
+int net_context_set_option(struct net_context *context,
```

Comment:
So that's basically like set_sockopt(), ok.

Reply:
Yes, I made it look similar so it is easy to map to setsockopt() if needed.